bitnami/redis/templates/redis-master-statefulset.yaml

```
@@ -162,6 +162,9 @@ spec:
             - name: REDIS_PORT
               value: {{ .Values.redisPort | quote }}
             {{- end }}
+            {{- if .Values.master.statefulset.extraEnv }}
+            {{- toYaml .Values.master.statefulset.extraEnv | trim | nindent 12 }}
```

We usually make use of a helper here that takes care of the correct rendering.

```suggestion
            {{- include "redis.tplValue" (dict "value" .Values.master.extraEnvVars "context" $) | nindent 12 }}
```

I see this Chart does not have the helper included, you just need to add this to the `_helpers.tpl` file:
```yaml
{{/*
Renders a value that contains template.
Usage:
{{ include "redis.tplValue" (dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "redis.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

---

I added it, however I had to leave the `trim` function, because for the following example : 
```
master:
  statefulset:
    extraEnv:
    - name: name
      value: value
    - name: other_name
      valueFrom:
        fieldRef:
          fieldPath: fieldPath
```
A newline was added at the end